index.js

```diff
@@ -335,10 +338,11 @@
   var TypeRep = $.NullaryType(
     'sanctuary/TypeRep',
     function(x) {
-      return R.type(x) === 'Function' ||
-             (x != null &&
-              R.type(x.name) === 'String' &&
-              R.type(x.test) === 'Function');
```

Comment:
The three lines above allowed a `Type` value (such as `$.Number` or `S.MaybeType`) to be used as a type representative. I'm almost certain we should not support this.